druid-filtering-ext/src/main/java/bi/deep/filtering/ip/range/impl/MultiRangeIPFilterImpl.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package bi.deep.filtering.ip.range.impl;
+
+import bi.deep.filtering.common.IPRange;
+import bi.deep.filtering.common.PredicateFactory;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableSet;
+import inet.ipaddr.IPAddress;
+import inet.ipaddr.IPAddressString;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.druid.error.InvalidInput;
+import org.apache.druid.query.filter.ColumnIndexSelector;
+import org.apache.druid.query.filter.Filter;
+import org.apache.druid.query.filter.ValueMatcher;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.index.BitmapColumnIndex;
+
+public class MultiRangeIPFilterImpl implements Filter {
+    private final String column;
+    private final Set<IPRange> ipV4Ranges;
+    private final Set<IPRange> ipV6Ranges;
+    private final boolean ignoreVersionMismatch;
+
+    public MultiRangeIPFilterImpl(String column, Set<IPRange> ranges, @Nullable Boolean ignoreVersionMismatch) {
```

Comment:
ignoreVersionMismatch is not nullable as primitive is sent there in MultiRangeIPFilter.toFilter, let's use primitive here as well

Reply:
Done